pkg/ingester/ingester.go

```diff
@@ -3120,6 +3116,24 @@ func (i *Ingester) ShutdownHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// checkAvailable checks whether the ingester is available, and if it is not the case
+// returns an unavailableError error.
+// Using block store, the ingester is available only when it is in a Running state.
+// The ingester is not available when stopping to prevent any read or writes to the
+// TSDB after the ingester has closed them.
+func (i *Ingester) checkAvailable() error {
+	s := i.State()
+	i.lifecycler.State()
+	if s == services.Running {
+		return nil
+	}
+	return newUnavailableError(
```

Comment:
nit: We don't need to call `s.String()` here, we can store `s` in the new error.